### ISSUE TITLE
Enable Aurora Enhanced Monitoring on existing instances - "IAM role invalid or wrong perms", followed by "MonitoringRoleARN value is required"

### ISSUE THREAD

**Issue by ssm1th** [hashicorp/terraform#9435]
Similar issue as @deanmraz reported in [ #5455](https://github.com/hashicorp/terraform/issues/5455) I think.

Attempting to enable Enhanced Monitoring on existing Aurora instances initially throws the following error:

```
InvalidParameterValue: IAM role ARN value is invalid or does not include the required permissions for: ENHANCED_MONITORING
```

When attempting an apply again following that error, I'm getting:

```
InvalidParameterCombination: A MonitoringRoleARN value is required if you specify a MonitoringInterval value other than 0
```

I suspect the latter is being thrown because the first apply actually added `monitoring_role_arn` to tfstate.

Have tried using the existing AWS provided managed policy and attaching it to a Terraform created role (all within Terraform), and I've also tried creating my own policy as per #5455 and attaching that to a self made role too, but the behaviour is still the same. 

The addition of `monitoring_role_arn` to tfstate following the first error sounds like a bug to me, but does anybody have any ideas or suggestions on how to get around this and get it working?

Thanks.

**Comment by ssm1th** [comment]
Tested with v0.7.5 and 0.7.7, same behaviour with both.

**Comment by afoldesi-gds** [comment]
TL;DR: Call apply twice. Once for creating the policy & role, and once for assigning the role to the database instance.

----

Hitting the same issue with 0.7.9 and MySql RDS. I've set up my own policy and using depends_on tags all the way through.

The reference to the monitoring_role_arn is added to the tfstate file after the first error requiring permissions. 

As a result the subsequent apply calls will not (try to) add it, so the second error is generated.

Deleting the role & policy and subsequent apply _does not_ fix the tfstate.

Manually deleting the offending line from the tfstate resets the process to the beginning and the first error is thrown again. 

----

Workaround when you already have a database instance running and you don't want to drop it:
- Remove the role, policy, and the DB monitoring settings from the terraform script you may have already added. Call apply.
- Manually remove the monitoring_role_arn line from the tfstate file if it's there.
- Add back only the IAM role and the corresponding policy into terraform. Call apply.
- Add back the database monitoring arn and interval settings. Call apply.
- Enjoy the logs pouring into cloudwatch.
